portfolio/src/main/client/src/components/controls/theme-switcher.ts

```diff
@@ -0,0 +1,25 @@
+import '@style/index.dark.scss';
+import '@style/index.light.scss';
+import {htmlElement} from '@src/util/html';
+
+document.body.setAttribute('data-theme', 'light');
+
+/**
+ * Toggles the theme between light and dark.
+ */
+export function toggle(): void {
+  // sample because we don't want this to run again
+  // in response to the theme being changed, that leads
+  // to infinite loop
+  if (document.body.getAttribute('data-theme') === 'light') {
+    document.body.setAttribute('data-theme', 'dark');
+  } else {
+    document.body.setAttribute('data-theme', 'light');
+  }
+}
+
+export const ThemeSwitcher = (): HTMLElement => {
+  const btn: HTMLButtonElement = htmlElement`<button class='theme-switcher'>Switch theme</button>`;
```

Comment:
Keep lines under 80 characters. You can do:
```
  const btn: HTMLButtonElement =
    htmlElement`...`;
```

Reply:
Jess said earlier that the line length should be 100 characters. Should I change it to 80?

Reply:
Interesting. I've always heard it as 100 characters for HTML and other non-code files, then 80 characters for anything else. @jessicaslaughter WDYT?

Reply:
Sorry about the confusion, I think I'd made a comment on an HTML file but not indicated which language that was for. I agree with Nick's comment!